rmm/monitor/src/rmi/rec/handlers.rs

```diff
@@ -25,14 +26,7 @@ pub fn set_event_handler(mainloop: &mut Mainloop) {
             return;
         }
 
-        let params_ptr = HostPointer::<Params>::new(arg[2], mm);
-        let params = params_ptr.acquire();
-        let params = if let Some(val) = params {
-            val
-        } else {
-            error!("access Params fail");
-            return;
-        };
+        host_pointer!(params, Params, arg[2], mm, ret[0]);
```

Comment:
I think it's better to add `_or_ret`. Because the control flow is changed when host_pointer is None.
`host_pointer_mut_or_ret` too.

```suggestion
        host_pointer_or_ret!(params, Params, arg[2], mm, ret[0]);
```

---

Later, if we enhance error handling, below form would be better .

```rust
let params = host_pointer!(Params, arg[2], mm)?;
trace!("{:?}", *params);
```

Reply:
> let params = host_pointer!(Params, arg[2], mm)?;

Agree. I'll leave a todo comment to do it. (we can do it after modifying error handling as you said)